src/arch/metal.rs

```diff
@@ -370,8 +370,14 @@ impl Architecture for MetalArch {
         );
     }
 
-    unsafe fn sfence_vma() {
-        asm!("sfence.vma")
+    unsafe fn sfence_vma(rs1: Register, rs2: Register) {
```

Comment:
I'm not really sure about this part tho, it looks like `usize` could be accepted as register number (same order as x0, x1...) but I would be glad to have confirmation.

Reply:
Hmmm, reading the doc it seems the semantic is a bit complicated, because the register `x0` has a special meaning... In the cases where one of the registers is not `x0` we need to load the corresponding value in the virtual register into a physical one and pass the index of that register to the VMA fence. So no it's not really `usizes`, the CPU is indeed going to read the content of the register from my understanding.